src/main/java/com/kylenicholls/stash/parameterizedbuilds/conditions/BuildPermissionsCondition.java

```diff
@@ -0,0 +1,54 @@
+package com.kylenicholls.stash.parameterizedbuilds.conditions;
+
+import com.atlassian.bitbucket.auth.AuthenticationContext;
+import com.atlassian.bitbucket.permission.Permission;
+import com.atlassian.bitbucket.permission.PermissionService;
+import com.atlassian.bitbucket.repository.Repository;
+import com.atlassian.bitbucket.setting.Settings;
+import com.atlassian.bitbucket.user.ApplicationUser;
+import com.atlassian.plugin.spring.scanner.annotation.imports.ComponentImport;
+import com.kylenicholls.stash.parameterizedbuilds.helper.SettingsService;
+import com.kylenicholls.stash.parameterizedbuilds.item.Job;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.Map;
+
+public class BuildPermissionsCondition extends BaseCondition{
+
+    private final PermissionService permissionService;
+    private final AuthenticationContext authContext;
+
+    @Autowired
+    public BuildPermissionsCondition(@ComponentImport PermissionService permissionService, SettingsService service, AuthenticationContext authContext) {
+        super(service);
+        this.permissionService = permissionService;
+        this.authContext = authContext;
+    }
+
+    public boolean checkPermissions(Job job, Repository repository, ApplicationUser user){
+        try {
+            Permission permissionRequired = Permission.valueOf(job.getPermissions());
+            return this.permissionService.hasRepositoryPermission(user, repository, permissionRequired);
+        } catch (IllegalArgumentException ex) {
+            return true;
```

Comment:
Yes, although it might be unnecessary with the changes to SettingsService line 96 and 97. I'll do some quick testing and see if that catch block can be removed.